build_tools/github_actions/fetch_test_configurations.py

```diff
@@ -0,0 +1,94 @@
+"""
+This script determines what test configurations to run
+
+Required environment variables:
+  - THEROCK_BIN_DIR
+  - PLATFORM
+"""
+
+import json
+import os
+from configure_ci import set_github_output
+
+THEROCK_BIN_DIR = os.getenv("THEROCK_BIN_DIR")
+
+test_matrix = {
+    # BLAS tests
+    "rocblas": {
+        "name": "rocblas",
+        "artifact_flags": "--blas --tests",
+        "timeout": 5,
+        "executable_command": f"{THEROCK_BIN_DIR}/rocblas-test --yaml {THEROCK_BIN_DIR}/rocblas_smoke.yaml",
```

Comment:
We could also put these in scripts, to allow for multi-line commands and to make it easier to developers/contributors to run the same tests that CI runs, outside of github actions.

I think a deeper question will be how to handle sharding though, as that may need to happen up a level in the workflows. In those cases, we could start outside of this matrix generator script.